examples/correlation_matrices/sampler.cpp

```diff
@@ -89,6 +89,24 @@ void write_to_file(std::string filename, std::vector<PointType> const& randPoint
     std::cout.rdbuf(coutbuf);
 }
 
+bool is_correlation_matrix(const MT& matrix){
+	
+	//check if all the diagonal elements are ones
+	for(int i=0 ; i<matrix.rows() ; i++){
+		if(matrix(i,i) != 1.0){
```

Comment:
Can you do an epsilon-equality check here?
For example `if (std::abs(matrix(i,i) - 1.0) > tol)`
where `const tol = 1e-08`.

Reply:
Yes, I can, thanks for this! I will make the changes.